TypeChain-DApp demo repo (#41)
rsksmart/demo-code-snippets: typechain-dapp/README.md

# Using TypeChain on frontend and backend

> This project demonstrates how to use TypeChain typings in DApp development on Rootstock.

Reviewer: ```suggestion
This project demonstrates how to use TypeScript types, via TypeChain, in DApp development on Rootstock.
```

## Contents of the repo
- `chain` Hardhat smart contract development environment
  - `contracts/MeowToken.sol` - demo Solidity smart contract
  - `tasks/deploy.ts` token deployment Hardhat task that:
    - compiles smart contracts, generates Hardhat artifacts and TypeChain typings
    - deploys smart contracts to a desired network
    - copies TypeChain typings to the frontend folder
  - `hardhat.config.ts` Hardhat config file with Rootstock networks configuration
- `front` React boilerplate application

>   - `scr/App.ts` App main component that:

Reviewer: ```suggestion
  - `src/App.ts` App main component that:
```

    - imports `ethers.js` convenience library 
    - imports TypeChain typings and deployed token address, generated by Hardhat
    - connects to Metamask and wraps its provider with `ethers.js` provider
    - uses TypeChain generated classes to connect to deployed smart contract

>     - envokes smart contract function to query connected user balance

Reviewer: ```suggestion
    - invokes smart contract function to query connected user balance
```
